src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -285,13 +285,11 @@ private static Uri CreateConnectUrl(Uri url, NegotiationResponse negotiationResp
                     if (Input.TryRead(out var buffer))
                     {
                         _logger.LogDebug("Scheduling raising Received event.");
-                        var ignore = _eventQueue.Enqueue(() =>
+                        var ignore = _eventQueue.Enqueue(async () =>
```

Comment:
When the only task involved is the final thing you run, we can avoid all the boilerplate C# sets up for an async method by just rewriting this like so:

```csharp
var ignore = _eventQueue.Enqueue(() =>
{
    _logger.LogDebug("Raising Received event.");
    return Received?.Invoke(buffer);
});
```

This means you run "synchronously" up to the call to Invoke and then just invoke it and pass it's `Task` back to the caller.